pkg/bundle/package.go

```diff
@@ -3,18 +3,60 @@ package bundle
 import (
 	"archive/tar"
 	"compress/gzip"
+	"fmt"
 	"io"
+	"os"
 	"path/filepath"
 
+	"github.com/massdriver-cloud/massdriver-cli/pkg/common"
 	"github.com/massdriver-cloud/massdriver-cli/pkg/compress"
 )
 
-func PackageBundle(filePath string, buf io.Writer) error {
+func Package(b *Bundle, filePath string, buf io.Writer) error {
+	buildDir, err := os.MkdirTemp("", "bundle-build")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(buildDir)
+
+	allowList := getAllowList(b)
+	copyConfig := common.CopyConfig{
+		Allows:  allowList,
+		Ignores: common.FileIgnores,
+	}
+	srcDir := filepath.Dir(filePath)
+	stats, errCopy := common.CopyFolder(srcDir, buildDir, &copyConfig)
+	if errCopy != nil {
+		return errCopy
+	}
+
+	mbs := common.FileSizeMB(stats.FolderSize)
+	if mbs > common.MaxBundleSizeMB {
+		return fmt.Errorf("Bundle size exceeds maximum allowed size of %vMB", common.MaxBundleSizeMB)
+	}
+
+	return tarFolder(buildDir, buf)
+}
+
+func getAllowList(b *Bundle) []string {
+	allAllows := []string{}
```

Comment:
non-blocking educational nit: could be more efficient we know the  length of this slice apriori so could pre-allocate.
though this might be slightly less readable.
```suggestion
	allAllows := make([]string, len(common.FileAllows) + len(b.Steps))
	copy(allAllows, common.FileAllows)
	if b.Steps != nil {
	  for i, step := range b.Steps {
	    allAllows[len(common.FileAllows)+i] = step.Path
	  }
	}
```

Reply:
Since the number of elements are so low I think I'll favor the readability here but you make a great point.